scalameta/contrib/src/main/scala/scala/meta/contrib/Extract.scala

```diff
@@ -0,0 +1,21 @@
+package scala.meta.contrib
+
+/**
+  * Logical extraction of B from A.
+  *
+  * Meaning that this is supposed to replicate what the use thinks should happen.
```

Comment:
How about:

> The motivation for this typeclass is to extract values out of trees that could have been fields on the classes. However, these values are not included as fields on the tree nodes because of various reasons.